src/hotspot/share/opto/divnode.cpp

```diff
@@ -44,19 +44,19 @@
 
 //----------------------magic_int_divide_constants-----------------------------
 // Compute magic multiplier and shift constant for converting a 32 bit divide
-// by constant into a multiply/shift/add series. Return false if calculations
-// fail.
+// by constant into a multiply/shift series.
 //
 // Borrowed almost verbatim from Hacker's Delight by Henry S. Warren, Jr. with
 // minor type name and parameter changes.
-static bool magic_int_divide_constants(jint d, jint &M, jint &s) {
+
+static void magic_int_divide_constants(jint d, jlong &M, jint &s) {
```

Comment:
I would prefer to see a strategy for applying gtest to these magic* functions (all of them, I think that's six).

One way to go about it would simply make them extern and write the gtests accordingly.

That would make me much happier than seeing tests for 7, 13, and a couple other constants!

An example of what I mean by using gtest is in the fix for JDK-8291649 which also applied to C2 constant folding logic.

I think we need this kind of test for any non-trivial constant folding or strength reduction logic, at least moving forward.

If the static magic functions were moved into a header file (not required but surely the high road here) I'd call it javaArithmetic.hpp.  And I'd consider moving java_add etc into there as well.  Alternatively, the static magic functions could be moved into globalDefinitions.hpp which is surprising but perhaps less disruptive than making a new header file for Java arithmetic but not factoring in the existing functions.  I don't recommend making a header file just for this particular algorithm all by itself; I'd rather lump more "stuff" into a single header file.

(I mildly disagree with  count_leading_zeros and count_trailing_zeroes being all by their lonesomes, and would expect a lumpier design putting more such stuff into powerOfTwo.hpp or a similar place.  Naming and grouping is hard.)